test/utils.js

```diff
@@ -108,33 +106,25 @@ export const sleep = async (timeout = 200) =>
 /**
  * Wait for condition to be met
  * E.g. await waitForCondition(() => emailSendMessageSpy.callCount === 1)
- * @param {*} cond
+ * @param {boolean | Promise<boolean>} cond
  * @param {*} options: { timeout, delay }
  * @returns {Promise}
  */
-export const waitForCondition = (cond, options = { timeout: 10000, delay: 0, onFailure: null }) =>
-  new Promise(resolve => {
-    let hasConditionBeenMet = false;
-    setTimeout(() => {
-      if (hasConditionBeenMet) {
-        return;
-      }
-      options.onFailure?.();
-      assert.fail(`Timeout waiting for condition: ${cond.toString()}`);
-      throw new Error('Timeout waiting for condition', cond);
-    }, options.timeout || 10000);
-    const isConditionMet = () => {
-      hasConditionBeenMet = Boolean(cond());
-      debugWaitForCondition(options.tag, `Has condition been met?`, hasConditionBeenMet);
-      if (hasConditionBeenMet) {
-        return setTimeout(resolve, options.delay || 0);
-      } else {
-        return setTimeout(isConditionMet, options.step || 100);
-      }
-    };
-    isConditionMet();
-  });
+export const waitForCondition = async (cond, options = { timeout: 10000, onFailure: null }) => {
```

Comment:
I needed a way to pass an async condition, took the opportunity to simplify the helper with async/await

Reply:
Nice one